charts/shoot-core/components/charts/kube-proxy/templates/kube-proxy-init-script.yaml

```diff
@@ -0,0 +1,25 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kube-proxy-init-script
+  namespace: kube-system
+data:
+  init.sh: |
+    #!/bin/sh
+    OLD_KUBE_PROXY_MODE=$(cat /kube-proxy-mode);
+    if [ -z ${OLD_KUBE_PROXY_MODE} ] || [ ${OLD_KUBE_PROXY_MODE} = ${KUBE_PROXY_MODE} ]; then
+      echo ${KUBE_PROXY_MODE} >/kube-proxy-mode;
+      exit 0
+    fi;
+    {{- if semverCompare "< 1.17" .Values.kubernetesVersion }}
+    /hyperkube
+    {{- if semverCompare "< 1.15" .Values.kubernetesVersion }}
+    proxy
+    {{- else }}
+    kube-proxy
+    {{- end }}
+    {{- else }}
+    /usr/local/bin/kube-proxy
+    {{- end }} --v=2 --cleanup --proxy-mode=${OLD_KUBE_PROXY_MODE} || exit $?;
```

Comment:
`--proxy-mode=${OLD_KUBE_PROXY_MODE}` I think this parameter is obsolete.

Reply:
That's true but does not work properly with env variable.

Reply:
I wanted to say that you do not need 
```
--proxy-mode=${OLD_KUBE_PROXY_MODE}
```
when
```
--cleanup
```
 is set, do you?

Reply:
I tested it and it was only properly working when it was set. In theory cleanup should do both but when testing it always skipped `ipvs` cleanup, when not explicitly set.